public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -170,6 +222,19 @@ export class ApiKeysPage extends PureComponent<Props, any> {
                   </select>
                 </span>
               </div>
+              <div className="gf-form max-width-21">
+                <span className="gf-form-label">Seconds to live</span>
+                <Input
+                  type="text"
+                  className="gf-form-input"
+                  placeholder="1d"
+                  validationEvents={timeRangeValidationEvents}
+                  value={newApiKey.secondsToLive}
+                  onChange={evt =>
+                    this.onApiKeyStateUpdate(evt, ApiKeyStateProps.SecondsToLive, this.transformToSecondsToLive)
```

Comment:
There is a an issue with having the transform here:
![Kapture 2019-06-20 at 10 49 08](https://user-images.githubusercontent.com/1014802/59834912-16725c80-9349-11e9-9cc9-c26eb1a927be.gif)

This feels jarring and unexpected to me and works differently than the other place we have this. So I think it would be better to do the transform when saving key.

Also that means `Seconds to live` label and `1d` placeholder looks a bit off. Can we change it to `Time to live` and add tooltip saying you can use different time units there `d/m/s/y...`?

Reply:
> Also that means Seconds to live label and 1d placeholder looks a bit off. Can we change it to Time to live and add tooltip saying you can use different time units there d/m/s/y...?

do you mean also to remove the placeholder and add a tooltip instead?

Reply:
I think we can leave the placeholder as `1d` and just change the label and add the tooltip. That seems most consistent and informative to me.

Reply:
We have frontend utility that can translate a time range like 10s 5m 15d to milliseconds and vice versa (think they are still in kbn.ts not sure, on mobile now)

Think this should use a time range like that as we have that everywhere else. It can be stored in seconds to make backend easier.

Reply:
You are right https://github.com/grafana/grafana/blob/4b0ad174ff43ba8b629a1c6b3c75051ad7f7c6c0/public/app/core/utils/kbn.ts#L212 seems like exactly what is needed there. There are some almost similar helpers in grafana/ui but probably this should be migrated there too so it is easier to find (though in separate PR)